source/shared/core_stmt.cpp

```diff
@@ -821,7 +827,13 @@ bool core_sqlsrv_fetch( _Inout_ sqlsrv_stmt* stmt, _In_ SQLSMALLINT fetch_orient
         }
         // First time only
         if ( !stmt->fetch_called ) {
-            SQLSMALLINT has_fields = core::SQLNumResultCols( stmt TSRMLS_CC );
+            SQLSMALLINT has_fields;
+            if (stmt->columns_rows_obtained) {
+                has_fields = stmt->column_count;
+            } else {
+                has_fields = core::SQLNumResultCols( stmt TSRMLS_CC );
+            }
```

Comment:
See my previous suggestion. Thus, call SQLNumResultCols() only if column_count is equal to that negative constant